### ISSUE TITLE
FIBO schema.org and FIBO OWL alignment proposal

### ISSUE THREAD

**Issue by sdml** [edmcouncil/fibo#880]
- [Thing](http://pending.schema.org/Thing)
	- [Action](http://pending.schema.org/Action)
		- [TransferAction](http://pending.schema.org/TransferAction)
			- [MoneyTransfer](http://pending.schema.org/MoneyTransfer) -> in FIBO - [ElectronicFundsTransferService](https://spec.edmcouncil.org/fibo/ontology/FBC/FunctionalEntities/FinancialServicesEntities/ElectronicFundsTransferService)
	- [Intangible](http://pending.schema.org/Intangible)
		- [Service](http://pending.schema.org/Service)
			- [FinancialProduct](http://pending.schema.org/FinancialProduct) -> in FIBO - [FinancialProduct](https://spec.edmcouncil.org/fibo/ontology/FBC/ProductsAndServices/FinancialProductsAndServices/FinancialProduct)
				- [BankAccount](http://pending.schema.org/BankAccount) -> in FIBO - [BankAccount](https://spec.edmcouncil.org/fibo/ontology/FBC/ProductsAndServices/ClientsAndAccounts/BankAccount)
					- [DepositAccount](http://pending.schema.org/DepositAccount) -> in FIBO - [DepositAccount](https://spec.edmcouncil.org/fibo/ontology/FBC/ProductsAndServices/ClientsAndAccounts/DepositAccount)
				- [CurrencyConversionService](http://pending.schema.org/CurrencyConversionService) -> in FIBO - [CurrencyConversionService](https://spec.edmcouncil.org/fibo/ontology/IND/ForeignExchange/ForeignExchange/CurrencyConversionService)
				- [InvestmentOrDeposit](http://pending.schema.org/InvestmentOrDeposit) -> in FIBO - [InvestmentOrDepositAccount](https://spec.edmcouncil.org/fibo/ontology/FBC/ProductsAndServices/ClientsAndAccounts/InvestmentOrDepositAccount)
					- [BrokerageAccount](http://pending.schema.org/BrokerageAccount) -> in FIBO - [BrokerageAccount](https://spec.edmcouncil.org/fibo/ontology/FBC/ProductsAndServices/ClientsAndAccounts/BrokerageAccount)
					- [DepositAccount](http://pending.schema.org/DepositAccount) -> in FIBO - [DepositAccount](https://spec.edmcouncil.org/fibo/ontology/FBC/ProductsAndServices/ClientsAndAccounts/DepositAccount)
					- [InvestmentFund](http://pending.schema.org/InvestmentFund) -> in FIBO - [InvestmentService](https://spec.edmcouncil.org/fibo/ontology/FBC/FunctionalEntities/FinancialServicesEntities/InvestmentService)
				- [LoanOrCredit](http://pending.schema.org/LoanOrCredit) -> in FIBO - [LoanOrCreditAccount](https://spec.edmcouncil.org/fibo/ontology/FBC/ProductsAndServices/ClientsAndAccounts/LoanOrCreditAccount)
					- [CreditCard](http://pending.schema.org/CreditCard) -> **MISSING**
					- [MortgageLoan](http://pending.schema.org/MortgageLoan) -> in FIBO - [MortgageLoanContract](https://spec.edmcouncil.org/fibo/ontology/LOAN/LoanContracts/LoanCore/MortgageLoanContract)
				- [PaymentCard](http://pending.schema.org/PaymentCard) -> **MISSING**
					- [CreditCard](http://pending.schema.org/CreditCard) -> **MISSING**
				- [PaymentService](http://pending.schema.org/PaymentService) -> in FIBO - [PaymentService](https://spec.edmcouncil.org/fibo/ontology/FBC/FunctionalEntities/FinancialServicesEntities/PaymentService)
			- [StructuredValue](http://pending.schema.org/StructuredValue)
				- [ExchangeRateSpecification](http://pending.schema.org/ExchangeRateSpecification) -> in FIBO - [ExchangeRate](https://spec.edmcouncil.org/fibo/ontology/FND/Accounting/CurrencyAmount/ExchangeRate)
				- [MonetaryAmount](http://pending.schema.org/MonetaryAmount) -> in FIBO - [MonetaryAmount](https://spec.edmcouncil.org/fibo/ontology/FND/Accounting/CurrencyAmount/MonetaryAmount)
				- [RepaymentSpecification](http://pending.schema.org/RepaymentSpecification) -> in FIBO - [PrincipalRepaymentTerms](https://spec.edmcouncil.org/fibo/ontology/FBC/DebtAndEquities/Debt/PrincipalRepaymentTerms)

**Comment by trypuz** [comment]
@sdml, 2 years ago, @ElisaKendall proposed the following [dependency mapping](https://wiki.edmcouncil.org/display/SCHEMAORG/Dependency+map+between+schema.org+financial+terms+and+FIBO): 
- MoneyTransfer --> FIBO [ElectronicFundsTransferService](https://spec.edmcouncil.org/fibo/ontology/FBC/FunctionalEntities/FinancialServicesEntities/ElectronicFundsTransferService)
- MortgageLoan --> FIBO [MortgageLoanContract](https://spec.edmcouncil.org/fibo/ontology/LOAN/LoanContracts/LoanCore/MortgageLoanContract)

So, it seems that we are missing only CreditCard from this picture.

**Comment by ElisaKendall** [comment]
A CreditCardAccount would be a subclass of LoanOrCreditAccount, but a CreditCard is not.

If we add them, PaymentCard would be a child of DebtInstrument, and CreditCard would be a child of PaymentCard, along with DebitCard as a sibling of CreditCard

What's the priority for adding this?

**Comment by ElisaKendall** [comment]
Also, accounts are not products in FIBO, which should remain as it is.  The Schema.org hierarchy is somewhat messed up.  But we can map to the right things in FIBO in any case.

**Comment by trypuz** [comment]
> A CreditCardAccount would be a subclass of LoanOrCreditAccount, but a CreditCard is not.
> 
> If we add them, PaymentCard would be a child of DebtInstrument, and CreditCard would be a child of PaymentCard, along with DebitCard as a sibling of CreditCard
> 
> What's the priority for adding this?

@ElisaKendall, thank you for your comments. Dependency mapping between FIBO and schema.org concepts is the first step to move our FIBO schema.org extension from pending to production. IMO it is very important.

**Comment by rivettp** [comment]
@trypuz as far as I can see the things listed are no longer Pending in schema.org - what makes you think they are?
If they are no longer pending does that change the priority?

**Comment by rivettp** [comment]
With respect to the proposal, MoneyTransfer in schema.org refers to one specific transfer (transaction) from A to B on a specific date, and probably best maps tohttps://spec.edmcouncil.org/fibo/ontology/FND/ProductsAndServices/PaymentsAndSchedules/Payment. That's quite different from the FIBO MoneyTransferService which is at the product level, and corresponds to PaymentService in schema.org.

**Comment by rivettp** [comment]
As per @ElisaKendall 's comment on Account, LoanOrCredit in schema.org is a product (e.g. "Fixed 30 year loan offered by Wells Fargo at 3% APR"), whereas LoanAccount in FIBO is a specific loan to a specific person for a specific amount on a specific date.  And may be linked to the Loan Product via which is was sold (though that link is of largely historical interest since the loan contract will be self-contained).
It makes sense for schema.org to focus at the product level since their objective is to support markup of websites which will be advertising the products. No one will want to (or be legally allowed to) have the loan accounts for specific customers searchable from their web site. OTOH FIBO is more concerned with the account level for purposes of risks, cash flows etc.

**Comment by rivettp** [comment]
A broader question to help me review this - what is the FIBO use case/scenario we're trying to serve with this? What is the primary direction? How complete need the mapping be?
For example one scenario might be to take banking product info represented in FIBO and generate schema.org markup for the bank's website - though not sure how useful that would be.

**Comment by trypuz** [comment]
> @trypuz as far as I can see the things listed are no longer Pending in schema.org - what makes you think they are?
> If they are no longer pending does that change the priority?

Some of them are Pending: BrokerageAccount, ExchangeRateSpecification, InvestmentFund, MoneyTransfer, MortgageLoan, and RepaymentSpecification.

**Comment by trypuz** [comment]
> A broader question to help me review this - what is the FIBO use case/scenario we're trying to serve with this? What is the primary direction? How complete need the mapping be?
> For example one scenario might be to take banking product info represented in FIBO and generate schema.org markup for the bank's website - though not sure how useful that would be.

FIBO use case is simple: create a markup for the financial products of banks (using FIBO schema.org extension) and show how this semantic description helps in data analytics. 

POC: http://finances.makolab.com/ (see also slides 7, 42-44 of https://www.slideshare.net/sopekmir/rank-analyse-lead-search)

**Comment by trypuz** [comment]
> If we add them, PaymentCard would be a child of DebtInstrument, and CreditCard would be a child of PaymentCard, along with DebitCard as a sibling of CreditCard

@ElisaKendall, could we add these classes to FIBO? It will complete our dependency mapping. Once it is complete we can submit it to schema.org and update our documentation: https://schema.org/docs/financial.html.

**Comment by trypuz** [comment]
Hi @ElisaKendall! Are you planning to add PaymentCard and CreditCard to FIBO or you want me to do that? LMK

**Comment by ElisaKendall** [comment]
Let me talk with my client and I'll get back to you.  I developed a complete model for credit cards that are not specific to the client, and they said that I could contribute it back as time permits.  I'll confirm that and let you know.

**Comment by trypuz** [comment]
@ElisaKendall, couldn't we start with adding just 2 classes (PaymentCard and CreditCard) to FIBO? We could discuss the complete model later...

**Comment by rivettp** [comment]
@trypuz note that CreditCard and PaymentCard recently added into FIBO by PR #1159 represent actual pieces of plastic owned by someone: they are not the credit card products as apparently needed by schema.org. For those FIBO (now) has CardProduct with subclasses CreditCardProduct and DebitCardProduct (missing other types of payment card).
Though (as per my comments on the PR) there are significant gaps in the properties people might want to use to comparison shop for a new credit card product.

**Comment by trypuz** [comment]
Hi Pete, thank you for your comment. What I see [here](https://spec.edmcouncil.org/fibo/ontology/FBC/ProductsAndServices/CardAccounts/CreditCard) does not match what you wrote about CreditCard. A credit card is a contract (an agreement) in FIBO, so it's something more than just a piece of plastic. More properties will be added in the Q4 2020.

**Comment by trypuz** [comment]
Resolved by #1159. Thank you, @ElisaKendall! 
@sdml, can we close this issues - see [CreditCard](https://spec.edmcouncil.org/fibo/ontology/FBC/ProductsAndServices/CardAccounts/CreditCard)

**Comment by rivettp** [comment]
Whether it's plastic or a contract, my point is that it's for one specific customer, not the general product that world be searched for on a website. For that FIBO has Credit card product.
Having said that, one of my many PR comments was the lack of a sufficient distinction between the account/contract and the plastic that is really only just evidence of it.

**Comment by ElisaKendall** [comment]
I think there may be an open issue around the distinction between an offer and a product - some credit card offers are not necessarily codified in 'standard products', but maybe credit card product is sufficient for the moment.  There are a number of subclasses we could include - kinds of credit card products, ranging from cash back, to rewards, to loyalty cards, etc. and we don't currently have all of that.  I would not spend any time on it unless we have a council member that has a use case and wants that level of detail, though.

**Comment by sdml** [comment]
Great! PaymentCard and CreditCard are both in FIBO now. Thank you.

**Comment by trypuz** [comment]
So, the financial schema.org extension is aligned with FIBO now. @sdml and @ElisaKendall - thank you for your effort. Hoped through this comment to inform @mfhepp and @sopekmir of the progress made... I also propose we update the financial schema.org documentation at https://schema.org/docs/financial.html.

**Comment by rivettp** [comment]
@sdml please note that CreditCard in schema.org corresponds to CreditCardProduct in FIBO. CreditCard in FIBO is for a specific card account held by a specific person and with its own account number, security code and expiry date. 
Likewise PaymentCard in schema.org maps to PaymentCardProduct in FIBO.

However CreditCard in schema.org is itself confusing - it seems to be both an enumeration (with values such as http://purl.org/goodrelations/v1#AmericanExpress) - which correspond to CreditCardNetwork in FIBO; and actual service offerings (such as the example given of a Wells Fargo Secured Visa Credit Card). 

@trypuz and @ElisaKendall is this dichotomy something you looked into before determining that this alignment issue has been resolved?

Personally I find it embarrassing that the schema.org page https://schema.org/CreditCard, as it stands, says it's based on work in FIBO!

**Comment by trypuz** [comment]
FIBO and schema.org have different philosophies behind (e.g. polysemy is pretty common in schema.org; FIBO allows only for multiple inheritance) and each of them is built with different objectives in mind following different methodologies. @rivettp, do not expect we transfer a part of FIBO directly to schema.org; we have to compromise.

**Comment by sdml** [comment]
I'm aware that CreditCard in schema.org means something different than in FIBO. It's quite common in schema that classes have multiple meanings. For example [LocalBusiness](https://schema.org/LocalBusiness) is both an organization and a physical place, so CreditCard can be a financial product and a plastic item.

**Comment by trypuz** [comment]
To continue this discussion, please create a new issue.
